client/jetpack-cloud/components/guided-tour/index.tsx

```diff
@@ -10,16 +10,39 @@ import { preferencesLastFetchedTimestamp } from 'calypso/state/preferences/selec
 import './style.scss';
 
 export interface Tour {
-	targetId: string;
+	target: string;
 	title: string;
 	description: string;
+	popoverPosition?:
+		| 'top'
+		| 'top right'
+		| 'right'
+		| 'bottom right'
+		| 'bottom'
+		| 'bottom left'
+		| 'left'
+		| 'top left';
 }
 
 interface Props {
 	tours: Tour[];
 	preferenceName: string;
 }
 
+// Helper function to get an element, returning a promise that resolves when the element is rendered.
+const getElementAsync = ( target: string ): Promise< HTMLElement > =>
```

Comment:
There are instances where React still renders the target element while the component already started looking for it. This function helps us get the target element by waiting for it to be rendered before rendering the popovers.

Reply:
Is there ever a chance that the target element never shows up? And would this cause performance problems if we continually request new animation frames?